docs/guides/Package-authoring-best-practices.md

```diff
@@ -0,0 +1,177 @@
+
+Example: Copyright (c) Contoso 2020
+
+### Licensing
+
+✔️ DO specify a valid license for your package.
+> [!IMPORTANT]
+> A project without a license defaults to [exclusive copyright](https://choosealicense.com/no-permission/), making it legally impossible for other people to use.
+
+✔️ CONSIDER using a license expression (SPDX identifier).
+> License expressions are surfaced the most clearly to package consumers and make it more obvious if the license has changed with a new release.
+
+✔️ CONSIDER specifying an MIT license.
+> If you want your package as to be usable by as many consumers as possible, the MIT license is a very well known non-restrictive license.
+
+❌ DO NOT use the deprecated `LicenseUrl` metadataproperty.
+> This presents legal ambiguity as license changes at the URL will retroactively change the displayed license for previous package versions.
+
+### Project URL
+
+✔️ CONSIDER including a link to an associated project or company website.
+> Company website links, even if not specific to the package/project, still improve consumer trust.
+
+❌ DO NOT put a repository URL (GitHub, BitBucket, etc.) in the `Project URL` field, as consumers will expect to find it in the `Repository URL` field.
+
+### Icon
+
+✔️ CONSIDER including an icon with your package to help visually differentiate it. It's a relatively small addition that can improve perception of package quality.
+> Icons can be specific to individual packages or be a brand logo.
+
+✔️ DO use an image that is 128x128 and has a transparent background (PNG) for best viewing results.
+> NuGet will automatically scale your image to the client it is being displayed on.
+
+❌ DO NOT use the deprecated `IconUrl` metadata property.
+
+### Repository Type and URL
+
+✔️ CONSIDER including a repository type and URL to make it easy for package consumers to view sourecode and package documentation (only if public).
+
+✔️ CONSIDER setting up [Source Link](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink) to add source control metadata to your assemblies and NuGet package.
+> Source Link will automatically adds `Repository URL` and `Repository Type` to the package metadata. It also adds the specific commit associated with your package version.
+
+### Tags
+
+✔️ DO include several tags with key terms related to your package to enhance discoverability.
+> Tags are taken into account in NuGet's search algorithm and are especially helpful for terms that are not in the Package ID.
+
+### Release notes
+
+✔️ CONSIDER including release notes with each update describing what changes were made.
+> While there is no specific format required for release notes, we recommend including:
+>
+> 1. Breaking changes
+> 2. New features
+> 3. Bug fixes
+> 
+> If you already track release notes in your repo, you can also include a link to your release notes file.
+
+## Dependencies
```

Comment:
Customers using "generated" pack can't, but customers specifying their own nuspec file can create dependencies directly under the nuspec's dependency node, rather than creating a group per supported TFM. nuget will warn on pack when it detects tfm-specific assets and no dependency group for that tfm, but should we also have a note in here saying to do that?  But as mentioned, anyone who lets nuget generate the nuspec for them will never have this problem.

Reply:
This is great. 

I'd love to see a `DO NOT use the dependencies element`. 

Throw in a couple of `do not` for good measure.

Reply:
Would it be better to recommend packing using proj files (instead of .nuspec files)? To me, generating a package using a .nuspec file is now an "advanced" scenario.